### PR TITLE
Hw 7 sb security

### PR DESCRIPTION
**Pull request by DPotapov93**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore01/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(
+                        auth -> auth
+                                .requestMatchers("/api/auth/**", "/error", "/api/swagger-ui/**")
```
Comment (vovasalyha) [inline src/main/java/mate/academy/onlinebookstore01/config/SecurityConfig.java:34]: Add also `/v3/api-docs/**`

File: src/main/java/mate/academy/onlinebookstore01/dto/book/CreateBookRequestDto.java [hunk 2]
```diff
@@ -26,8 +26,8 @@ public class CreateBookRequestDto {
     private String author;
     @NotNull(message = ISBN_EXCEPTION)
     @Pattern(regexp
-            = "^(?:ISBN(?:-1[03])?:? )?(?=[0-9X]{10}$|(?=(?:[0-9]+[- ]){3})\n"
-            + "[- 0-9X]{13}$|97[89][0-9]{10}$|(?=(?:[0-9]+[- ]){4})[- 0-9]{17}$)\n"
+            = "^(?:ISBN(?:-1[03])?:? )?(?=[0-9X]{10}$|(?=(?:[0-9]+[- ]){3})"
```
Comment (vovasalyha) [inline src/main/java/mate/academy/onlinebookstore01/dto/book/CreateBookRequestDto.java:29]: Try to avoid adding unrelated changes to PRs

File: src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            tableName: users_roles
+            columns:
+              - column:
+                  name: user_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+                    primaryKey: true
+                    deleteCascade: true
+                    foreignKeyName: user_foreignKey
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml:16]: Try to name you constraints so it is globally unique. Here you can see a pattern for foreign keys: https://repository.prace-ri.eu/git/help/development/database/constraint_naming_convention.md

File: src/main/resources/db/changelog/changes/08-insert-users-roles-table.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+databaseChangeLog:
+  - changeSet:
+      id: insert-users-roles-table
+      author: dmytro
+      changes:
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/08-insert-users-roles-table.yaml:5]: try to improve your code and use valueComputed with preConditions for your insert.
Example:
```
      preConditions:
        - sqlCheck:
            expectedResult: 1
            sql: SELECT COUNT(*) FROM users WHERE email='admin@example.com'
        - sqlCheck:
            expectedResult: 1
            sql: SELECT COUNT(*) FROM roles WHERE name='ROLE_ADMIN'
      changes:
        - insert:
            tableName: user_role
            columns:
              - column:
                  name: user_id
                  valueComputed: (SELECT id FROM users WHERE email='admin@example.com')
              - column:
                  name: role_id
                  valueComputed: (SELECT id FROM roles WHERE name='ROLE_ADMIN')
```

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

File: src/main/java/mate/academy/onlinebookstore01/dto/book/CreateBookRequestDto.java (hunk 2 above)
Comment (d1sam) [reply src/main/java/mate/academy/onlinebookstore01/dto/book/CreateBookRequestDto.java:29]: still exists, you can fix pattern issue with isbn in another pr.

File: src/main/java/mate/academy/onlinebookstore01/model/User.java [hunk 5]
```diff
@@ -24,4 +32,48 @@ public class User {
...
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/model/User.java:62]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/onlinebookstore01/model/User.java [hunk 6]
```diff
@@ -24,4 +32,48 @@ public class User {
...
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return true;
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/model/User.java:67]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/onlinebookstore01/model/User.java [hunk 7]
```diff
@@ -24,4 +32,48 @@ public class User {
...
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/model/User.java:72]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/onlinebookstore01/repository/RoleRepository.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package mate.academy.onlinebookstore01.repository;
+
+import mate.academy.onlinebookstore01.model.Role;
+import org.springframework.data.jpa.repository.JpaRepository;
+
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore01/repository/RoleRepository.java:5]:
```suggestion
@Repository
```

File: src/main/resources/db/changelog/changes/03-create-roles-table.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: name
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/03-create-roles-table.yaml:20]:
```suggestion
                    nullable: false
                    unique: true
```

File: src/main/resources/db/changelog/changes/03-create-roles-table.yaml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-roles-table
+      author: dmytro
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/03-create-roles-table.yaml:4]:
```suggestion
      author: dmytro_potapov
```
it's better to provide full name as author of changeSet

File: src/main/resources/db/changelog/changes/04-insert-roles.yaml [hunk 11]
```diff
@@ -0,0 +1,17 @@
+databaseChangeLog:
+  - changeSet:
+      id: insert-roles
+      author: dmytro
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/04-insert-roles.yaml:4]:
```suggestion
      author: dmytro_potapov
```
it's better to provide full name as author of changeSet

File: src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml [hunk 12]
```diff
@@ -0,0 +1,26 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-users-roles-table
+      author: dmytro
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/05-create-users-roles-table.yaml:4]:
```suggestion
      author: dmytro_potapov
```
it's better to provide full name as author of changeSet

File: src/main/resources/db/changelog/changes/06-add-isDeleted-column-to-users-table.yaml [hunk 13]
```diff
@@ -0,0 +1,14 @@
+databaseChangeLog:
+  - changeSet:
+      id: add-is-deleted-to-users
+      author: dmytro
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/06-add-isDeleted-column-to-users-table.yaml:4]:
```suggestion
      author: dmytro_potapov
```
it's better to provide full name as author of changeSet

File: src/main/resources/db/changelog/changes/07-insert-users-table.yaml [hunk 14]
```diff
@@ -0,0 +1,21 @@
+databaseChangeLog:
+  - changeSet:
+      id: insert-users-table
+      author: dmytro
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/07-insert-users-table.yaml:4]:
```suggestion
      author: dmytro_potapov
```
it's better to provide full name as author of changeSet

File: src/main/resources/db/changelog/changes/08-insert-users-roles-table.yaml [hunk 15]
```diff
@@ -0,0 +1,30 @@
+databaseChangeLog:
+  - changeSet:
+      id: insert-users-roles-table
+      author: dmytro
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/08-insert-users-roles-table.yaml:4]:
```suggestion
      author: dmytro_potapov
```
it's better to provide full name as author of changeSet

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but some mistakes are still exist.

**Review by vovasalyha — APPROVED** [review]
